### PR TITLE
Remove unary plus in DSLs

### PR DESCRIPTION
**Pull request by seelchen**
This removes all unary plus operators for the DSL classes. The objects created in the functions are instead added to the lists automatically.

### CODE REVIEW

File: codyze-backends/cpg/src/test/kotlin/de/fraunhofer/aisec/codyze/backends/cpg/NfaDfaConstructionTest.kt [hunk 1]
```diff
@@ -208,17 +208,15 @@ class NfaDfaConstructionTest {
...
-                                +TestClass::start
-                                +TestClass::process.maybe()
-                                +TestClass::finish
-                            }
-                        )
+                some {
+                    TestClass::create or
+                        group {
+                            +TestClass::start
+                            TestClass::process.maybe()
```
Comment (maximilian-galanis) [inline codyze-backends/cpg/src/test/kotlin/de/fraunhofer/aisec/codyze/backends/cpg/NfaDfaConstructionTest.kt:215]: Having to use '+' on standalone `OrderTokens` but not on `OrderTokens` on which we call something e.g., `TestClass::process.maybe()` might be confusing. Perhaps we should change it to be `maybe(TestClass::process)`.
Then the behavior would be more predictable:
- no need for '+' when calling any method
- add a '+' for standalone `OrderTokens`

Comment (maximilian-galanis) [reply codyze-backends/cpg/src/test/kotlin/de/fraunhofer/aisec/codyze/backends/cpg/NfaDfaConstructionTest.kt:215]: Additionally, we could make the `add` method public. Then the user would not have to use '+' anywhere, but still has the option to do so.
Having the `add` method publicly available should also help with intellij autocomplete.

File: codyze-specification-languages/coko/coko-dsl/src/test/resources/javaimpl.codyze.kts [hunk 2]
```diff
@@ -4,9 +4,9 @@ plugins { id("cpg") }
 
 class JavaLogging : Logging {
     override fun log(message: String, vararg args: Any) = op {
-        +definition("java.util.logging.Logger.info") {
-            +signature {
-                +group {
+        definition("java.util.logging.Logger.info") {
+            signature {
+                group {
                     +message
```
Comment (maximilian-galanis) [inline codyze-specification-languages/coko/coko-dsl/src/test/resources/javaimpl.codyze.kts:10]: We could give the user the option to use the `add` method here too

**Review by maximilian-galanis — CHANGES_REQUESTED** [review]

File: codyze-backends/cpg/src/test/kotlin/de/fraunhofer/aisec/codyze/backends/cpg/NfaDfaConstructionTest.kt (hunk 1 above)
Comment (seelchen) [reply codyze-backends/cpg/src/test/kotlin/de/fraunhofer/aisec/codyze/backends/cpg/NfaDfaConstructionTest.kt:215]: > Having to use '+' on standalone `OrderTokens` but not on `OrderTokens` on which we call something e.g., `TestClass::process.maybe()` might be confusing. Perhaps we should change it to be `maybe(TestClass::process)`. Then the behavior would be more predictable:
> 
>     * no need for '+' when calling any method
> 
>     * add a '+' for standalone `OrderTokens`

That is true, the suggestions sounds good! We already have `maybe(vararg tokens)` as a function which covers `maybe(TestClass::process)` so I will just delete `OrderToken.maybe()` and the corresponding functions for the other quantifiers.

**Review by maximilian-galanis — APPROVED** [review]
Looks great! Once all checks pass, you can merge it.

**Pull request merged**